build/org.eclipse.pde.build/src/org/eclipse/pde/internal/build/Utils.java

```diff
@@ -884,18 +839,18 @@ static public StringBuffer readFile(InputStream stream) throws IOException {
 	 *
 	 */
 	public static void updateVersion(File buildFile, String propertyName, String version) throws IOException {
-		StringBuffer buffer = readFile(buildFile);
-		int pos = scan(buffer, 0, propertyName);
+		StringBuilder buffer = new StringBuilder(Files.readString(buildFile.toPath(), Charset.defaultCharset()));
```

Comment:
Is a stringbuilder actually neded here?

Reply:
note that JDKs also offers an two valued indexOf for String.
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html#indexOf(java.lang.String,int)

Reply:
> Is a stringbuilder actually neded here?

Thinking about it, not really. At least not until the very end.
Does anybody know a more elegant way to replace a substring based on indices?

Btw. I also traced who is writing the file and seems to the AntScript class, which uses UTF-8, so we can just the no-argument overload.
https://github.com/eclipse-pde/eclipse.pde/blob/c8754be8f6e23e9c5aebf6bf11b59515e98d262d/build/org.eclipse.pde.build/src/org/eclipse/pde/internal/build/ant/AntScript.java#L50C56-L50C78

Reply:
great.